### PR TITLE
Implement non-nested struct statistic extraction

### LINKED ISSUE #10609: Incorrect statistics read for struct array in parquet
### Describe the bug

I found this while adding tests https://github.com/apache/datafusion/pull/10608. The statistics of struct array returns nothing

### To Reproduce

See `test_struct` in https://github.com/apache/datafusion/pull/10608

### Expected behavior

Return some values for the statistics

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Lordworms**
## Which issue does this PR close?

Closes #10609

### CODE REVIEW

**Comment by Lordworms** [comment]
Did not implement the nested struct type since the current RowGroupMetaData and ColumnChunkMetaData seems not to support nesting
image
image

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 1]
```diff
@@ -1155,7 +1155,6 @@ async fn test_boolean() {
 // BUG
 // https://github.com/apache/datafusion/issues/10609
 // Note that: since I have not worked on struct array before, there may be a bug in the test code rather than the real bug in the code
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:1157]: perhaps we can update the comments here too

Comment (Lordworms) [reply datafusion/core/tests/parquet/arrow_statistics.rs:1157]: Got it

File: datafusion/core/src/datasource/physical_plan/parquet/statistics.rs [hunk 2]
```diff
@@ -190,20 +190,29 @@ pub(crate) fn parquet_column<'a>(
...
-        // to correctly walk the parquet schema accounting for the
-        // logical type rules - <https://github.com/apache/parquet-format/blob/master/LogicalTypes.md>
-        //
-        // For example a ListArray could correspond to anything from 1 to 3 levels
-        // in the parquet schema
-        return None;
-    }
+    match field.data_type() {
+        DataType::Struct(_) => {
+            let parquet_idx = (0..parquet_schema.columns().len())
```
Comment (alamb) [inline datafusion/core/src/datasource/physical_plan/parquet/statistics.rs:195]: @tustvold  could you help here -- is this the correct way to find the correct parquet leaf from an arrow struct type?

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 3]
```diff
@@ -1168,7 +1165,7 @@ async fn test_struct() {
         reader: reader.build().await,
         expected_min: Arc::new(struct_array(vec![(Some(1), Some(6.0), Some(12.0))])),
         expected_max: Arc::new(struct_array(vec![(Some(2), Some(8.5), Some(14.0))])),
-        expected_null_counts: UInt64Array::from(vec![0]),
+        expected_null_counts: UInt64Array::from(vec![2]),
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:1168]: Something isn't right here -- the structs themselves aren't nullable (only some of their child fields)

https://github.com/apache/datafusion/blob/7c08a6fd00fcb352c11889ed62de7c9948978c79/datafusion/core/tests/parquet/mod.rs#L954-L958

https://github.com/apache/datafusion/blob/7c08a6fd00fcb352c11889ed62de7c9948978c79/datafusion/core/tests/parquet/mod.rs#L1024-L1045

File: datafusion/core/src/datasource/physical_plan/parquet/statistics.rs [hunk 4]
```diff
@@ -216,6 +225,96 @@ pub(crate) fn min_statistics<'a, I: Iterator<Item = Option<&'a ParquetStatistics
...
+/// Extract the nullcount statistics for struct array
+pub(crate) fn struct_null_count_statistics(
+    row_groups: &[RowGroupMetaData],
+    struct_fields: &Fields,
+) -> Result<ArrayRef, DataFusionError> {
+    if struct_fields.iter().any(|f| f.data_type().is_nested()) {
+        return Ok(Arc::new(new_empty_array(&DataType::UInt64)) as ArrayRef);
+    }
+
+    let mut null_count: u64 = 0;
```
Comment (alamb) [inline datafusion/core/src/datasource/physical_plan/parquet/statistics.rs:306]: I think nulls only have to check the current nulls (not their children's nulls)...

**Review by alamb — COMMENTED** [review]
Thank you @Lordworms

File: datafusion/core/src/datasource/physical_plan/parquet/statistics.rs (hunk 4 above)
Comment (Lordworms) [reply datafusion/core/src/datasource/physical_plan/parquet/statistics.rs:306]: sure, I'll fix it

**Pull request closed without merging**